### ISSUE TITLE
Autoencoders and Deep Neural Network

### ISSUE THREAD

**Issue by hwang-happy** [Shark-ML/Shark#136]
hi sharker, thanks for your help, shark library was correctly installed on my computer, thank you !
Some question about deep learning trouble me.

when i build NN

**method 1 ,** 
CrossEntropy loss;
ErrorFunction error(dataset,&network,&loss);
initRandomUniform(network,-0.1,0.1);
IRpropPlus optimizer;
error.init();
optimizer.init(error);
unsigned numberOfSteps = 1000;
for(unsigned step = 0; step != numberOfSteps; ++step)
        optimizer.step(error);

**method2**
CrossEntropy loss;
ErrorFunction error(dataset,&network,&loss);
**TwoNormRegularizer regularizer(error.numberOfVariables());
error.setRegularizer(regularisation,&regularizer);**
initRandomUniform(network,-0.1,0.1);
IRpropPlus optimizer;
error.init();
optimizer.init(error);
unsigned numberOfSteps = 1000;
for(unsigned step = 0; step != numberOfSteps; ++step)
        optimizer.step(error);

so i want ti know what's the different between those tow method since method2 include a regularizer, where can i find a detail Tutorial which can Explain those concepts

**Comment by Ulfgard** [comment]
Hi,

the difference is that regularization penalizes large weights in your network, which is one of many ways to prevent overfitting and behaviour that does not generalize well to new networks data points.
This special regularization is covered in most of the material covering the basics of machine learning and neural networks, often termed "weight decay".
